packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -550,70 +547,78 @@ export class CapabilityService {
     return subscribedPrices;
   }
 
+  /**
+   * Fetch the mergedConfig of plans that are configured and subscribed to.
+   */
+  private async configuredSubscribedMergedConfigs(subscribedPrices: string[]) {
+    if (!this.paymentConfigManager) return [];
+    const allPlans = (await this.paymentConfigManager.allPlans()).filter(
+      (plan) => subscribedPrices.includes(plan.stripePriceId ?? '')
+    );
+    // Can't use .map here because TS is unable to infer that paymentConfigManager
+    // is not undefined even though it is checked above as of TS 5.2.2.
```

Comment:
Kinda silly that you can't use map but using paymentConfigManager inside a for...of is fine.

Reply:
It's because it's after the async call, and the reference could be undefined by the time this code runs. AFAIK TypeScript is correctly throwing an error here, covering a case that could absolutely occur.

Reply:
If you mean the async call on line 555, isn't the reference on line 562 also after it?

Reply:
Yea, I was a bit surprised that it couldn't infer that the narrowing above was still valid at this point as no async has happened.

Reply:
It's not going to matter much as the whole block of code is being removed soon. We're going to be fetching config from Contentful, not Firestore, which is what this bit is doing.